lib/MySQL_Protocol.cpp

```
@@ -276,8 +304,39 @@ void MySQL_Protocol::generate_one_byte_pkt(unsigned char b) {
+	assert((*myds) != NULL);
+	assert(data != NULL || data_len == 0);
+	assert(data_len <= 0xFFFFFFU - 1);
+
+	uint8_t sequence_id = (*myds)->pkt_sid + 1;
+	mysql_hdr myhdr;
+	myhdr.pkt_id = sequence_id;
+	myhdr.pkt_length = static_cast<uint32_t>(data_len + 1);
+
+	const unsigned int size = myhdr.pkt_length + sizeof(mysql_hdr);
+	unsigned char *_ptr = static_cast<unsigned char *>(l_alloc(size));
+	memcpy(_ptr, &myhdr, sizeof(mysql_hdr));
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)